src/ui/libs/DatalensChartkit/components/Control/Items/Items.js

```diff
@@ -76,28 +86,36 @@ function BaseControlSelect({
             key: value,
         }));
 
-    const wrappedValue = multiselect ? wrapToArray(value) : value;
-
     const wrappedOnChange = React.useCallback(
         (value) => {
             // null - when SINGLE with allowEmptyValue and there is no selected value, otherwise [null] is obtained in the parameters
             // [] - when MULTIPLE and there is no selected value, otherwise results in the parameters []
             const wrappedValue =
                 value === null || (Array.isArray(value) && !value.length) ? '' : value;
+
+            setCurrentValue(multiselect ? wrapToArray(wrappedValue) : wrappedValue);
             return onChange(wrappedValue);
         },
-        [onChange],
+        [onChange, multiselect],
     );
 
+    const isRequired = Utils.isEnabledFeature(Feature.SelectorRequiredValue) && !isValueRequired;
+    const showSelectAll =
+        Utils.isEnabledFeature(Feature.SelectorRequiredValue) &&
+        currentValue?.length === items?.length &&
+        isValueRequired
+            ? false
+            : undefined;
+
     const size = isMobileView ? MOBILE_SIZE.YC_SELECT : 's';
 
     return (
         <YCSelect
             showSearch={searchable}
             type={multiselect ? YCSelect.MULTIPLE : YCSelect.SINGLE}
-            allowEmptyValue={true}
+            allowEmptyValue={isRequired}
```

Comment:
Is it okay that when feature is disabled we will not allow empty values?
`const isRequired = Utils.isEnabledFeature(Feature.SelectorRequiredValue) && !isValueRequired`
in that case if `Utils.isEnabledFeature(Feature.SelectorRequiredValue)` is false, `allowEmptyValue` will be false as well,